experiments/AtmosGCM/moist-baroclinic-wave-bulksfcflux.jl

```diff
@@ -0,0 +1,414 @@
+    τ_int_1::FT = A * (τ_z_1 - 1) + B * z * τ_z_3
+    τ_int_2::FT = C * z * τ_z_3
+    I_T::FT =
+        (cos(φ) * (1 + γ * z / _a))^k -
+        k / (k + 2) * (cos(φ) * (1 + γ * z / _a))^(k + 2)
+
+    # base state virtual temperature, pressure, specific humidity, density
+    T_v::FT = (τ_1 - τ_2 * I_T)^(-1)
+    p::FT = _p_0 * exp(-_grav / _R_d * (τ_int_1 - τ_int_2 * I_T))
+
+    # base state velocity
+    U::FT =
+        _grav * k / _a *
+        τ_int_2 *
+        T_v *
+        (
+            (cos(φ) * (1 + γ * z / _a))^(k - 1) -
+            (cos(φ) * (1 + γ * z / _a))^(k + 1)
+        )
+    u_ref::FT =
+        -_Ω * (_a + γ * z) * cos(φ) +
+        sqrt((_Ω * (_a + γ * z) * cos(φ))^2 + (_a + γ * z) * cos(φ) * U)
+    v_ref::FT = 0
+    w_ref::FT = 0
+
+    # velocity perturbations
+    F_z::FT = 1 - 3 * (z / z_t)^2 + 2 * (z / z_t)^3
+    if z > z_t
+        F_z = FT(0)
+    end
+    d::FT = _a * acos(sin(φ) * sin(φ_c) + cos(φ) * cos(φ_c) * cos(λ - λ_c))
+    c3::FT = cos(π * d / 2 / d_0)^3
+    s1::FT = sin(π * d / 2 / d_0)
+    if 0 < d < d_0 && d != FT(_a * π)
+        u′::FT =
+            -16 * V_p / 3 / sqrt(3) *
+            F_z *
+            c3 *
+            s1 *
+            (-sin(φ_c) * cos(φ) + cos(φ_c) * sin(φ) * cos(λ - λ_c)) /
+            sin(d / _a)
+        v′::FT =
+            16 * V_p / 3 / sqrt(3) * F_z * c3 * s1 * cos(φ_c) * sin(λ - λ_c) /
+            sin(d / _a)
+    else
+        u′ = FT(0)
+        v′ = FT(0)
+    end
+    w′::FT = 0
+    u_sphere = SVector{3, FT}(u_ref + u′, v_ref + v′, w_ref + w′)
+    u_cart = sphr_to_cart_vec(bl.orientation, u_sphere, aux)
+
+    ## Compute ure profile
+    ## Pressure coordinate η
+    ## η_crit = p_t / p_w ; p_t = 10000 hPa, p_w = 340 hPa
+    η = p / _p_0
+    if η > η_crit
+        q_tot = q_0 * exp(-(φ / φ_w)^4) * exp(-((η - 1) * _p_0 / p_w)^2)
+    else
+        q_tot = q_t
+    end
+    phase_partition = PhasePartition(q_tot)
+
+    ## temperature & density
+    T::FT = T_v / (1 + M_v * q_tot)
+    ρ::FT = air_density(bl.param_set, T, p, phase_partition)
+
+    ## potential & kinetic energy
+    e_pot::FT = gravitational_potential(bl.orientation, aux)
+    e_kin::FT = 0.5 * u_cart' * u_cart
+    e_tot::FT = total_energy(bl.param_set, e_kin, e_pot, T, phase_partition)
+
+    ## Assign state variables
+    state.ρ = ρ
+    state.ρu = ρ * u_cart
+    state.ρe = ρ * e_tot
+
+    if bl.moisture isa EquilMoist
+        state.moisture.ρq_tot = ρ * q_tot
+    end
+
+    nothing
+end
+
+"""
+Defines analytical function for prescribed T_sfc and q_sfc, following
+Thatcher and Jablonowski (2016), used to calculate bulk surface fluxes.
+
+T_sfc_pole = SST at the poles (default: 271 K), specified above
+"""
+struct Varying_SST_TJ16{PS, O, MM}
+    param_set::PS
+    orientation::O
+    moisture::MM
+end
+function (st::Varying_SST_TJ16)(state, aux, t)
+    FT = eltype(state)
+    φ = latitude(st.orientation, aux)
+
+    _T_sfc_pole = T_sfc_pole(st.param_set)::FT
```

Comment:
@vchuravy: if you change the line above to:
```julia
_T_sfc_pole::FT = T_sfc_pole(st.param_set)
```
Then you get a GPU compilation error for a non-`isbits` type. It isn't related just to callable types because the same thing will happen if you change the assignment to `_C_drag` in line 241 below.